src/optimization/budget.py

```
@@ -0,0 +1,292 @@
+"""예산 최적화 — Task 3.1 & 3.2 (배한나).
+
+Uplift Score + CLV를 결합하여 제한된 예산 내 ROI를 극대화하는
+고객별 마케팅 예산 배분을 산출한다.
+
+방법론
+------
+1. 그리디 알고리즘: priority score(uplift×0.5 + CLV×0.3 + 세그먼트×0.2) 기준 내림차순 정렬 후 예산 소진까지 선택
```

Ruff (0.15.13)

[warning] Docstring contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF002)